k2/k2/reports/junit/test/test_writer.py

```diff
@@ -138,7 +153,6 @@ def test_run_verdict_mapped_to_test_case(self):
 def no_exception(time_index, verdict, name_suffix=''):
     return dedent(
         f"""\
-        <?xml version="1.0" encoding="utf-8"?>
```

Comment:
what caused the removal of the XML prolog?

Reply:
The `xmldiff` library would give me an exception if I left it there. I believe I can restore it and sprinkle the code with `encode('utf-8')` instead. It seems those are the 2 options that xmldiff give me:
* give it `bytes()` with encoding tag, or
* give it `str()` without encoding tag.

I didn't know which one was best, so I chose one of them.

Side note: it seems the _junit_ report writer produce a `str()`, while the _testng_ report writer produce a `bytes()`. Again, I don't know which one is most correct. Feels like it's out of scope for this pull request at least.

Reply:
Prolog restored in latest diff, and strings encoded to python `bytes()` objects instead